app/components/SearchViewControl.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+import React from 'react';
+import SearchActionCreators from '../actions/SearchActionCreators';
+
+class SearchViewControl extends React.Component {
+  handleClick(e) {
+    SearchActionCreators.selectView(e.currentTarget.dataset.view);
+  }
+
+  render() {
+    let className = 'view-selector';
+    if (this.props.selected) {
+      className += ' selected';
+    }
+    return (
+      <button
+        className={className}
+        data-view={this.props.view}
+        onClick={this.handleClick}
+      >
+        {this.props.children}
+      </button>
+    );
+  }
+}
+
+SearchViewControl.propTypes = {
+  children: React.PropTypes.element.isRequired,
```

Comment:
Yes it is. And I validate it only with element because I didn't find any React.children like validator, and it only receives one element (< i >)